webapp/packages/core-blocks/src/FormControls/InputFile.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { observer } from 'mobx-react-lite';
+import { ReactNode, useCallback, useContext, useState } from 'react';
+import styled, { css, use } from 'reshadow';
+
+import type { ComponentStyle } from '@cloudbeaver/core-theming';
+import { blobToData, bytesToSize } from '@cloudbeaver/core-utils';
+
+import { Button } from '../Button';
+import type { ILayoutSizeProps } from '../Containers/ILayoutSizeProps';
+import { IconButton } from '../IconButton';
+import { useTranslate } from '../localization/useTranslate';
+import { UploadArea } from '../UploadArea';
+import { useStyles } from '../useStyles';
+import { baseFormControlStyles, baseInvalidFormControlStyles, baseValidFormControlStyles } from './baseFormControlStyles';
+import { FormContext } from './FormContext';
+
+const DEFAULT_MAX_FILE_SIZE = 2048;
+
+const INPUT_FILE_FIELD_STYLES = css`
+  field-label {
+    display: block;
+    composes: theme-typography--body1 from global;
+    font-weight: 500;
+  }
+  field-label:not(:empty) {
+    padding-bottom: 10px;
+  }
+  field-description {
+    display: flex;
+    align-items: center;
+    gap: 2px;
+  }
+  IconButton {
+    width: 12px;
+    height: 12px;
+    flex-shrink: 0;
+    cursor: pointer;
+    &:hover {
+      opacity: 0.5;
+    }
+  }
+`;
+
+interface Props<TState> extends ILayoutSizeProps {
+  name: keyof TState;
+  state: TState;
+  labelTooltip?: string;
+  tooltip?: string;
+  required?: boolean;
+  style?: ComponentStyle;
+  /** Max file size in KB */
+  maxFileSize?: number;
+  disabled?: boolean;
+  className?: string;
+  children?: ReactNode;
+  onChange?: (value: string, name: keyof TState) => void;
+}
+
+type InputFileType = <TState extends Record<string, any>>(props: Props<TState>) => React.ReactElement<any, any>;
+
+export const InputFile: InputFileType = observer(function InputFile({
+  name,
+  state,
+  labelTooltip,
+  tooltip,
+  required,
+  small,
+  medium,
+  large,
+  tiny,
+  style,
+  maxFileSize = DEFAULT_MAX_FILE_SIZE,
+  disabled,
+  className,
+  children,
+  onChange,
+}: Props<Record<any, any>>) {
+  const translate = useTranslate();
+  const context = useContext(FormContext);
+
+  const [selected, setSelected] = useState<File | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  const styles = useStyles(
+    INPUT_FILE_FIELD_STYLES,
+    baseFormControlStyles,
+    style,
+    error ? baseInvalidFormControlStyles : baseValidFormControlStyles
+  );
+
+  const fileSaved = !!state[name];
+  const description = error?.message ?? selected?.name ?? translate(fileSaved ? 'ui_processing_saved' : 'ui_no_file_chosen');
+
+  const removeFile = useCallback(() => {
```

Comment:
don't use useCallback (it's too simple component, without passing props through many components, if it changed it probably will be fully rerender)